Formula/mozjpeg.rb

```diff
@@ -1,15 +1,28 @@
 class Mozjpeg < Formula
   desc "Improved JPEG encoder"
   homepage "https://github.com/mozilla/mozjpeg"
-  url "https://github.com/mozilla/mozjpeg/releases/download/v3.1/mozjpeg-3.1-release-source.tar.gz"
-  sha256 "deedd88342c5da219f0047d9a290cd58eebe1b7a513564fcd8ebc49670077a1f"
+  revision 1
 
-  head do
-    url "https://github.com/mozilla/mozjpeg.git"
+  stable do
+    url "https://github.com/mozilla/mozjpeg/releases/download/v3.1/mozjpeg-3.1-release-source.tar.gz"
+    sha256 "deedd88342c5da219f0047d9a290cd58eebe1b7a513564fcd8ebc49670077a1f"
 
-    depends_on "autoconf" => :build
-    depends_on "automake" => :build
-    depends_on "libtool" => :build
+    # Fix crashes in ASM code with current llvm/clang versions.
+    # See https://github.com/libjpeg-turbo/libjpeg-turbo/pull/20
+    # and https://github.com/mozilla/mozjpeg/issues/202 for info.
+    patch do
+      url "https://patch-diff.githubusercontent.com/raw/libjpeg-turbo/libjpeg-turbo/pull/20.diff"
+      sha256 "d29a4b48fff3d3f22d1281f4125ff2bdf84eb25819facd724a8096ae04e6cd94"
+    end
+
+    # Fix negative shift with IFAST FDCT and qual=100
+    # Patch from freebsd/freebsd-ports based on commit:
+    # https://github.com/libjpeg-turbo/libjpeg-turbo/commit/4cfa3f4
+    # See https://github.com/mozilla/mozjpeg/pull/186 for info.
+    patch :p0 do
```

Comment:
The patch isn't in HEAD either, so even if we did jump the gun and accept this early, why only apply it to stable?